### DISCUSSION TITLE
Producing Jobs from Applications Outside of Rust

### DISCUSSION THREAD

**Discussion by c12i** [apalis-dev/apalis#391] [Q&A]
I am currently exploring the possibility of producing jobs from applications written in languages other than Rust. Specifically, I am interested in understanding if it's feasible to create jobs in the queue from non-Rust environments while maintaining compatibility with apalis's processing mechanisms.

Questions
- **Compatibility**: Is it possible to produce jobs from non-Rust applications, such as those written in Python, JavaScript, or any other language, while ensuring that these jobs are correctly handled by Apalis?
- **Job Format**: If it's possible, could you provide guidance or documentation on the required job format and the specific Redis data structures that need to be populated? This would be crucial for ensuring interoperability.

Any insights would be greatly appreciated.

**Comment by geofmureithi** [comment]
Hey, its pretty much possible esp with the recent 0.6 version.
There are a few things to consider:
1. Codec: Ensure the same codec is used on both sides eg json
2. Backend: Different backends store items differently, refer to the push/schedule implementations.

For example in redis, please refer to the `push_jobs.lua` script and the `push` method.
The job format is here: https://github.com/geofmureithi/apalis/blob/353d6f8450e862daa3553cb953a73a15bdd9cea4/packages/apalis-redis/src/storage.rs#L72-L74

I intend to work on the `Request` struct to improve on this. Ref #393 

That should get you started. Let me know if you have any questions, or thoughts.
